### ISSUE TITLE
Better error messages when loading the wrong Wasm binary?

### ISSUE THREAD

**Issue by surma** [emscripten-core/emscripten#12796]
The [Squoosh](https://squoosh.app) spent quite a long time why were were getting the following error:

```
Compress-9e039c93.js:1 Uncaught (in promise) TypeError: Cannot read property 'apply' of undefined
    at o.___wasm_call_ctors (features-worker-18f82dab.js:1)
	[...]
```

It turns out, we forgot to adjust a path and were using the glue code from one Emscripten module with the Wasm binary from another Emscripten module:

```js
import jxlEncoder, { JXLModule } from 'codecs/jxl/enc/jxl_enc';
import wasmUrl from 'url:codecs/webp/enc/webp_enc.wasm';
```

I wish Emscripten would give me a better error message. 

**Implementation idea:** Would it be worth embedding a small data string containing a UID in the Wasm binary to check that the glue code is loading the .wasm file it was designed for?

**Comment by sbc100** [comment]
Sounds reasonable to me.   Would you imagine this being enable via an option?  Or enabled by default in debug builds perhaps?  Or enabled by default in all build (even `-Oz`)?

**Comment by surma** [comment]
I am not sure. At first I thought that tying this to `-s ASSERTIONS=1` would make sense, but then it wouldn’t have actually helped in our scenario (at least not until we recompiled). 

Maybe a (more) helpful error message could already be generated when expected .wasm exports are absent, and that would be enabled in all builds imo. But I’ll defer to y’all here.

**Comment by sbc100** [comment]
Yes that is also a good idea.  I'd like to make that happen if we can find a way to do it without effecting codesize too much (simply including the list of expected exports could do this if we are not careful).
